File: Addition_android_app/app/src/main/java/com/paresh/addition/MainActivity.java
package com.paresh.addition;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        final TextView tv3 = (TextView)findViewById(R.id.tv3);
        final EditText et1 = (EditText)findViewById(R.id.et1);
        final EditText et2 = (EditText)findViewById(R.id.et2);
        Button add = (Button)findViewById(R.id.add);


        final float[] ans = new float[1];

        add.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                ans[0] = Float.parseFloat(et1.getText().toString()) + Float.parseFloat(et2.getText().toString());
                tv3.setText(String.valueOf(ans[0]));
            }
        });

    }
}

File: android_apps/SnapChatClone/app/src/main/java/com/example/snapchatclone/MainActivity.kt
package com.example.snapchatclone

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.EditText
import android.widget.Toast
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.database.FirebaseDatabase
import com.google.firebase.ktx.Firebase

class MainActivity : AppCompatActivity() {
    var userNameEditText:EditText?=null
    var emailEditText : EditText?=null
    var passwordEditText : EditText?=null
    val mAuth = FirebaseAuth.getInstance()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        userNameEditText = findViewById(R.id.userNameEditText)
        emailEditText = findViewById(R.id.emailEditText);
        passwordEditText = findViewById(R.id.passwordEditText)

        if(mAuth.currentUser != null){
            login()
        }
    }

    fun goClicked(view: View){
        //Check if we can login the user

        mAuth.signInWithEmailAndPassword(emailEditText?.text.toString(), passwordEditText?.text.toString())
            .addOnCompleteListener(this) { task ->
                if (task.isSuccessful) {
                    login();
                } else {
                    // Sign up user
                    mAuth.createUserWithEmailAndPassword(emailEditText?.text.toString(), passwordEditText?.text.toString())
                        .addOnCompleteListener(this) { task ->
                            if(task.isSuccessful){
                                //Add to database
                                FirebaseDatabase.getInstance().getReference().child("users").child(task.result!!.user?.uid!!).child("email").setValue(emailEditText?.text.toString())
                                FirebaseDatabase.getInstance().getReference().child("users").child(task.result!!.user?.uid!!).child("userName").setValue(userNameEditText?.text.toString())
                                login()
                            }else{
                                Toast.makeText(this,"Login failed. Try again", Toast.LENGTH_SHORT).show()
                            }
                        }
                }
            }
    }

    fun login(){
        //Move to next activity
        val intent = Intent(this, SnapsActivity::class.java)
        startActivity(intent)
    }
}